implementations/c/ockam/log/CMakeLists.txt

```diff
@@ -1,18 +1,40 @@
-
 # ---
-# ockam::log_interface
+# ockam::log
 # ---
-add_library(ockam_log_interface INTERFACE)
-add_library(ockam::log_interface ALIAS ockam_log_interface)
+add_library(ockam_log)
+add_library(ockam::log ALIAS ockam_log)
 
 set(INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/include)
 
-target_include_directories(ockam_log_interface INTERFACE ${INCLUDE_DIR})
+target_include_directories(ockam_log PUBLIC ${INCLUDE_DIR})
+
+message(STATUS ${INCLUDE_DIR})
 
 file(COPY log.h DESTINATION ${INCLUDE_DIR}/ockam)
 
 target_sources(
-  ockam_log_interface
-  INTERFACE
+  ockam_log
+  PRIVATE
     ${INCLUDE_DIR}/ockam/log.h
-)
\ No newline at end of file
+    log.c
+)
+
+target_link_libraries(
+  ockam_log
+  PUBLIC
+    ockam::error_interface
+)
+
+if (OCKAM_DISABLE_LOG)
+  message(STATUS "Log is disabled")
+  add_definitions(-DOCKAM_DISABLE_LOG)
```

Comment:
I suspect this may need to be tied to the target, with `target_compile_definitions ` maybe.

Reply:
@mrinalwadhwa tnx for pointing out. I was expecting target_add_definitions to exist, but I was wrong. Fixed